docs/en/advanced_guides/engine.md

```diff
@@ -1 +1,276 @@
+| :-------------------------------------------------------------------------------------------------------------------: | :--------------------------------------------------------------------------------------------: | :----------: |
+|                [EMAHook](https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/ema_hook.py)                 |              apply Exponential Moving Average (EMA) on the model during training.              | NORMAL (50)  |
+|         [EmptyCacheHook](https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/empty_cache_hook.py)         |            release all unoccupied cached GPU memory during the process of training.            | NORMAL (50)  |
+|        [SyncBuffersHook](https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/sync_buffer_hook.py)         | synchronize model buffers such as running_mean and running_var in BN at the end of each epoch. | NORMAL (50)  |
+| [NaiveVisualizationHook](https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/naive_visualization_hook.py) |               Show or Write the predicted results during the process of testing.               | LOWEST (100) |
+
+### Hooks implemented in MMSelfsup
+
+Some hooks have been already implemented in MMSelfsup, they are:
+
+- [DeepClusterHook](mmselfsup.engine.hooks.DeepClusterHook)
+
+- [DenseCLHook](mmselfsup.engine.hooks.DenseCLHook)
+
+- [ODCHook](mmselfsup.engine.hooks.ODCHook)
+
+- [SimSiamHook](mmselfsup.engine.hooks.SimSiamHook)
+
+- [SwAVHook](mmselfsup.engine.hooks.SwAVHook)
+
+- ......
+
+An example:
+
+Take [DenseCLHook](https://github.com/open-mmlab/mmselfsup/blob/dev-1.x/mmselfsup/engine/hooks/densecl_hook.py) for example, this hook includes `loss_lambda` warmup in DenseCL.
+
+`loss_lambda` is loss weight for the single and dense contrastive loss. Defaults to 0.5.
+
+```python
+losses = dict()
+losses['loss_single'] = loss_single * (1 - self.loss_lambda)
+losses['loss_dense'] = loss_dense * self.loss_lambda
+```
+
+`DenseCLHook` is implemented as follows:
+
+```python
+...
+@HOOKS.register_module()
+class DenseCLHook(Hook):
+...
+    def before_train_iter(self,
+                          runner,
+                          batch_idx: int,
+                          data_batch: Optional[Sequence[dict]] = None) -> None:
+...
+        cur_iter = runner.iter
+        if cur_iter >= self.start_iters:
+            get_model(runner.model).loss_lambda = self.loss_lambda
+        else:
+            get_model(runner.model).loss_lambda = 0.
+
+```
+
+If the hook is already implemented in MMEngine or MMSelfsup, you can directly modify the config to use the hook as below
+
+```python
+custom_hooks = [
+    dict(type='MMEngineHook', a=a_value, b=b_value, priority='NORMAL')
+]
+```
+
+such as using `EMAHook`, start_iters is 500:
+
+```python
+custom_hooks = [
+    dict(type='DenseCLHook', start_iters=500)
+]
+```
+
+## Optimizer
+
```

Comment:
Add one description here to we introduce `Optimizer` section through 3 different parts